File: python/27d_490A_TeamOlympiad.py
def main():
    w = int(input())
    array = input().split()
    col = [[],[],[]]
    for i in range(w):
        array[i] = int(array[i])
        col[array[i] - 1].append(i + 1)
    numb = min(len(col[0]), min(len(col[1]), len(col[2])))
    if numb != 0:
        print(numb)
        ans = []
        while numb > 0:
            t1 = col[0][-1]
            col[0].pop()
            t2 = col[1][-1]
            col[1].pop()
            t3 = col[2][-1]
            col[2].pop()
            ans.append([t1, t2, t3])
            numb -= 1
        for i in ans:
            for j in i:
                print(j, end = " ")
            print()
    else:
        print(numb)

if __name__ == "__main__":
    main()

File: cpp/27d_490A_TeamOlympiad.cpp
#include<bits/stdc++.h>

int main(){
    int n, temp;
    std::vector<std::vector<int>> arrays(3);
    std::cin>>n;
    for(int i = 0; i < n; i++){
        std::cin>>temp;
        arrays[temp - 1].push_back(i + 1);
    }
    int numb = std::min(arrays[0].size(), std::min(arrays[1].size(), arrays[2].size()));
    if (numb != 0){
        std::cout<<numb<<"\n";
        int t1, t2, t3;
        std::vector<std::vector<int>> ans;
        while(numb --){
            t1 = arrays[0].back();
            arrays[0].pop_back();
            t2 = arrays[1].back();
            arrays[1].pop_back();
            t3 = arrays[2].back();
            arrays[2].pop_back();
            ans.push_back({t1, t2, t3});
        }
        for(int i = 0; i < ans.size(); i++){
            for(auto j : ans[i]){
                std::cout<<j<<" ";
            }
            std::cout<<"\n";
        }
    }
    else{
        std::cout<<numb;
    }
    return 0;
}